src/mono/mono/mini/debugger-agent.c

```diff
@@ -9212,8 +8786,19 @@ frame_commands (int command, guint8 *p, guint8 *end, Buffer *buf)
 		return ERR_ABSENT_INFORMATION;
 
 	switch (command) {
+	case CMD_STACK_FRAME_GET_ARGUMENT: {
+		pos = decode_int (p, &p, end);
+		if (sig->hasthis) {
+			if (pos == 0)
+				goto cmd_stack_frame_get_this;
+			else
+				pos--;
+		}
+		len = 1;		
+		header = mono_method_get_header_checked (frame->actual_method, error);
+		goto cmd_stack_frame_get_parameter;
```

Comment:
consider (but it's up to you) refactoring the code for `cmd_stack_frame_get_this` and `cmd_stack_frame_get_parameter` into separate utility functions.

Reply:
done.